zookeeper-api/src/main/java/org/apache/helix/zookeeper/zkclient/ZkClient.java

```diff
@@ -384,6 +405,21 @@ public void unsubscribeStateChanges(
   }
 
   public void unsubscribeAll() {
+    if (_usePersistWatcher) {
```

Comment:
I am getting a bit confused. if we have recursive watcher and persist watcher, this code will change, correct?

Reply:
persist watcher only refers to non recursive data/child watcher. 
Same as data/child watcher, recursive watcher will has its own registry, we will need to clean that registry as well here, but it won't change current impl.